src/conversions.jl

```diff
@@ -140,9 +140,10 @@ pyptr_query(po::PyObject) = pyisinstance(po, c_void_p_Type) || pyisinstance(po,
 # typealias PyAny Union{PyObject, Int, Bool, Float64, Complex128, AbstractString, Function, Dict, Tuple, Array}
 abstract PyAny
 
-pyany_toany(T::Type) = T
-pyany_toany(T::Type{PyAny}) = Any
-pyany_toany(T::Type{Vararg{PyAny}}) = Vararg{Any}
+function pyany_toany(T::Type)
+    T === Vararg{PyAny} ? Vararg{Any} : T
+end
+pyany_toany(::Type{PyAny}) = Any
```

Comment:
The usage of it has always been strange. It is essentially only meaningful in a tuple but it's hard to only allow it there without breaking many other things. It is used to be allowed to use it just like a type though almost all types constructed this way are invalid. This is not disallowed.

It is still allowed to be used as a value and is still treated as a type for dispatch purpose even if it never actually was one. I believe the long term goal is to get rid of it or somehow being able to pass it around without dealing with this special case or generating invalid types.